src/migrations/simple_migration.js

```diff
@@ -0,0 +1,158 @@
+import Migration from './migration';
+const DEFAULT_TYPE = 'doc';
+
+/**
+ * SimpleMigration requires you to declare _savedObjectType OR _searchQuery, _newVersion, _index (optional), _type (optional),
+ * in constructor and two methods: _shouldObjectUpgrade and _upgradeObject which would be passed the objects by
+ * the base class to count and upgrade objects if the version is < _newVersion.
+ *
+ * Warning: Every new migration should bump the saved object's Version (defined by _newVersion, positive integer).
+ */
+export default class SimpleMigration extends Migration {
+  constructor(configuration) {
+    super(configuration);
+    this._defaultIndex = configuration.config.get('kibana.index');
+  }
+
+  /**
+   * @return {number}
+   */
+  get newVersion() {
+    if (!this._newVersion || !(parseInt(this._newVersion) > 0)) {
+      throw new Error('Every SimpleMigration must declare a new object version (positive integer).');
+    }
+    return parseInt(this._newVersion);
+  }
+
+  get scrollOptions() {
+    return this._scrollOptions || {
+      size: 100
+    };
+  }
+
+  /**
+   * Returns the number of objects that can be upgraded by this migration.
+   * @return {number}
+   */
+  async count() {
+    const { index, type, query } = this._getSearchParams();
+    const objects = await this.scrollSearch(index, type, query);
+    let count = 0;
+    for (let i = 0; i < objects.length; i++) {
```

Comment:
simplify + I am not sure but await might not work correctly in regular loop
but it WILL work in **for of**

```javascript
for (const savedObject of objects) {
  if (await this.checkOutdated(savedObject)) {
    count++;
  }
}
```